### DISCUSSION TITLE
Have any idea to turn radio input  value to boolean?

### DISCUSSION THREAD

**Discussion by wildfrontend** [react-hook-form/react-hook-form#1903] [Ideas] [answered]
it does look like the input radio type value alway return string , even value setted to boolean type
```
<input type="radio"   value={false} />
```
so when submitting form data, it will get string type

is any idea to change value type from react-hook-form ?

**Comment by bluebill1049** [comment, marked as answer]
This is native input behavior.

however you can use Controller to parse value: https://codesandbox.io/s/react-hook-form-parse-and-format-textarea-furtc

**Reply by wildfrontend** [reply]
Oh~ yeah !
i can change type by onChange event  thanks

**Comment by dandv** [comment]
@wildfrontend: sounds just like my earlier question about [transforming the value returned by a controller](https://github.com/react-hook-form/react-hook-form/discussions/1717).

**Reply by bluebill1049** [reply]
i think it's pretty easy with Controller after V6: https://codesandbox.io/s/controller-parse-input-value-q52si?file=/src/App.tsx we wouldn't need more complicated props.
